src/foam/dao/AbstractDAO.js

```diff
@@ -592,7 +601,14 @@ if ( ( skip > 0 ) && ( skip < AbstractDAO.MAX_SAFE_INTEGER ) ) {
 }
 
 if ( order != null ) {
-  sink = new OrderedSink(order, null, sink);
+  var innerSink = sink;
+  while ( innerSink instanceof ProxySink ) {
+    innerSink = ((ProxySink) innerSink).getDelegate();
+  }
+
+  if ( ! ( innerSink instanceof foam.mlang.sink.Count ) ) {
```

Comment:
I think you can only do this if the outer sink is a CountSink. Otherwise, you don't need to worry about it. Your original code was correct.